botocore/signers.py

```diff
@@ -86,7 +87,7 @@ def sign(self, operation_name, request):
             signature_version=signature_version, request_signer=self)
 
         # Sign the request if the signature version isn't None or blank
-        if signature_version:
+        if signature_version != botocore.UNSIGNED:
```

Comment:
Will this affect the CLI? We set the endpoint's signature version to ``None``. You can check real quick by running this test: https://github.com/aws/aws-cli/blob/develop/tests/integration/customizations/s3/test_plugin.py#L1594